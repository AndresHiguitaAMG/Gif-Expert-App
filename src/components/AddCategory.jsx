import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
    const [ inputValue, setInputValue ] = useState("");
    
    const handleOnInputChange = e => {
        setInputValue(e.target.value);
    };

    const handleOnSubmit = e => {
        e.preventDefault();
        if (inputValue.trim().length <= 1) return;
        onNewCategory(inputValue.trim());
        setInputValue("");
    };

    return (
        <form onSubmit={handleOnSubmit}>
            <input
                type={'text'}
                placeholder={'Buscar gifs'}
                value={inputValue}
                onChange={handleOnInputChange}
            />
            <button type={'submit'}>Agregar</button>
        </form>
  );
};

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
};