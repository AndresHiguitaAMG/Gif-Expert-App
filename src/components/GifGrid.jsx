import useFetchGifs from '../hooks/useFetchGifs';
import GifItem from './GifItem';

const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
        <h3>{category}</h3>
        <div className='card-grid'>
          {
            images.map((i) => (
              <GifItem 
              key={i.id}
              {...i}
              />
            ))
          }
        </div>
    </>
  );
};

export default GifGrid;