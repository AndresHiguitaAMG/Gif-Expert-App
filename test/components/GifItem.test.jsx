import { render } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Prubeas en <GifItem', () => {
    test('debe hacer match con el snapshot', () => {
        const url = 'https://giphy.com/clips/cameronpaulsmith-bH03afFirwWm0Qn6wo';
        const title = "Pikachu";
        const { container } = render(<GifItem url={url} title={title} />);

        expect(container).toMatchSnapshot();
    });
});