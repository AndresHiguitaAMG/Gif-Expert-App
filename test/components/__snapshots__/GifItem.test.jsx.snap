// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubeas en <GifItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Pikachu"
      src="https://giphy.com/clips/cameronpaulsmith-bH03afFirwWm0Qn6wo"
    />
    <p>
      Pikachu
    </p>
  </div>
</div>
`;
